.github/workflows/unit-test-on-pull-request.yml

```diff
@@ -42,14 +42,13 @@ jobs:
         uses: ./.github/workflows/env
         with:
           target_arch: ${{ matrix.target_arch }}
-      - name: Get linter version
-        id: linter-version
-        run: (echo -n "version="; make linter-version) >> "$GITHUB_OUTPUT"
       - name: linter
         env:
           GOARCH: ${{ matrix.target-arch }}
           CGO_ENABLED: 1
-        run: make lint
+        run: |
+          make lint
+          git diff --exit-code || (echo 'Code base is not Go formatted, please run "make lint" and commit the changes in this PR.' && exit 1)
```

Comment:
Can we add this to a separate step (see my other comment below) so that we don't conflate linting with formatting?